sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -230,9 +236,9 @@ class DynamicPartitionDataWriter(
       description.customPartitionLocations.get(PartitioningUtils.parsePathFragment(dir))
     }
     val currentPath = if (customPath.isDefined) {
-      committer.newTaskTempFileAbsPath(taskAttemptContext, customPath.get, ext)
+      committer.newTaskTempFileAbsPath(taskAttemptContext, customPath.get, prefix, ext)
```

Comment:
We need the `prefix` parameter because we want to put the bucket id at the beginning of the file name?

Reply:
@cloud-fan - yes, this is the cleanest way I can think of to achieve that (compatible file naming), let me know if you have any other idea, thanks.